### DISCUSSION TITLE
Custom integration no longer maintained

### DISCUSSION THREAD

**Discussion by gerard33** [custom-components/climate.e_thermostaat#22] [General]
This component is no longer maintained as I don't own an ICY e-thermostaat anymore. I can still merge PRs made and tested by the community and release new versions.
It was really fun to develop this custom integration 🌞

**Comment by lbuijk** [comment]
**Anyone a solution?**

Logger: homeassistant.helpers.entity
Bron: helpers/entity.py:1655

Entity None (<class 'custom_components.e_thermostaat.climate.EThermostaat'>) is using deprecated supported features values which will be removed in HA Core 2025.1. Instead it should use <ClimateEntityFeature.TARGET_TEMPERATURE|PRESET_MODE: 17>, please create a bug report at https://github.com/custom-components/climate.e_thermostaat/issues and reference https://developers.home-assistant.io/blog/2023/12/28/support-feature-magic-numbers-deprecation

Logger: homeassistant.components.climate.const
Bron: helpers/deprecation.py:222

CURRENT_HVAC_HEAT was used from e_thermostaat, this is a deprecated constant which will be removed in HA Core 2025.1. Use HVACAction.HEATING instead, please report it to the author of the 'e_thermostaat' custom integration
CURRENT_HVAC_IDLE was used from e_thermostaat, this is a deprecated constant which will be removed in HA Core 2025.1. Use HVACAction.IDLE instead, please report it to the author of the 'e_thermostaat' custom integration
HVAC_MODE_HEAT was used from e_thermostaat, this is a deprecated constant which will be removed in HA Core 2025.1. Use HVACMode.HEAT instead, please report it to the author of the 'e_thermostaat' custom integration
SUPPORT_PRESET_MODE was used from e_thermostaat, this is a deprecated constant which will be removed in HA Core 2025.1. Use ClimateEntityFeature.PRESET_MODE instead, please report it to the author of the 'e_thermostaat' custom integration
SUPPORT_TARGET_TEMPERATURE was used from e_thermostaat, this is a deprecated constant which will be removed in HA Core 2025.1. Use ClimateEntityFeature.TARGET_TEMPERATURE instead, please report it to the author of the 'e_thermostaat' custom integration

**Comment by gerard33** [comment]
Can you copy the `climate.py` version from the PR #23  in your `custom_components/e_thermostaat` folder and restart HA to see if the issues are fixed?

**Reply by lbuijk** [reply]
Nope, still not working. I made a workaround with a lot of cURL commands which do a POST en GET to en from the ICY cloud...fine for me.

**Reply by Josh004** [reply]
Hi, How did you solve this?

**Reply by lbuijk** [reply]
Not a solve but a workaround:

In configuration.yaml

shell_command:
zet_temp_thermostaat: /usr/bin/curl -s -X POST -H "Session-token:$(/usr/bin/curl -s --request POST 'https://portal.icy.nl/login' --data 'username=USERNAME&password=PASSWORD' | awk -F":" '{print $13}' | awk -F'"' '{print $2}')" -d "temperature1=$(/usr/bin/curl -s -X GET -H "Authorization:Bearer INLOG TOKEN HOME ASSISTANT" -H "Content-Type:application/json" 'https://IP HOME ASSISTANT/api/states/input_number.zet_thermostaat_op' -k | awk -F":" '{print $3}' | awk -F'"' '{print $2}')" -d "uid=UID" --data 'username=USERNAME&password=PASSWORD' 'https://portal.icy.nl/data'

(in image because it gives in the comment an error while it works and it is good....)
![image](https://github.com/user-attachments/assets/866dc822-391f-40d7-93d0-ca264af3f286)


lees_temp_thermostaat: /usr/bin/curl -s -H "Session-token:$(/usr/bin/curl -s --request POST 'https://portal.icy.nl/login' --data 'username=USERNAME&password=PASSWORD' | awk -F":" '{print $13}' | awk -F'"' '{print $2}')" --request GET 'https://portal.icy.nl/data' --data 'USERNAME&password=PASSWORD' | awk -F"," '{print $6}' | awk -F':' '{print $2}'

lees_zet_thermostaat: /usr/bin/curl -s -H "Session-token:$(/usr/bin/curl -s --request POST 'https://portal.icy.nl/login' --data 'username=USERNAME&password=PASSWORD' | awk -F":" '{print $13}' | awk -F'"' '{print $2}')" --request GET 'https://portal.icy.nl/data' --data 'USERNAME&password=PASSWORD' | awk -F"," '{print $5}' | awk -F':' '{print $2}'

What this does (and maybe it can be more simple....this is my way):
- get a session token so you can put a new temp in the ICY cloud. The new temp is a variable in home assistant (see below)
- read the actual temp from the thermostat
- read the set temp from the thermostat

Automations:
**ICY-Read:**
alias: ICY - Read
description: ""
triggers:
  - trigger: time_pattern
    minutes: /1
    enabled: true
conditions: []
actions:
  - action: shell_command.lees_temp_thermostaat
    data: {}
    response_variable: output1
  - action: input_number.set_value
    metadata: {}
    data:
      value: "{{output1.stdout}}"
    target:
      entity_id: input_number.lees_temp_thermometer
  - action: shell_command.lees_zet_thermostaat
    data: {}
    response_variable: output2
  - action: input_number.set_value
    metadata: {}
    data:
      value: "{{output2.stdout}}"
    target:
      entity_id: input_number.zet_thermostaat_op
mode: single

**ICY-Set**
alias: ICY - Set
description: ""
triggers:
  - trigger: state
    entity_id:
      - input_number.zet_thermostaat_op
    from: null
    to: null
conditions: []
actions:
  - action: shell_command.zet_temp_thermostaat
    metadata: {}
    data: {}
mode: single

**Number box card:**
type: custom:numberbox-card
border: true
entity: input_number.zet_thermostaat_op
secondary_info: "Huidig: %input_number.lees_temp_thermometer:state °C"
min: 5
max: 35
step: 0.5


![image](https://github.com/user-attachments/assets/703dfeff-48dc-41a8-bd90-649bc8696899)

**Reply by Josh004** [reply]
Hi, How do I get a ICY INLOG TOKEN?

**Reply by Josh004** [reply]
I have already managed to read them temp  (both)! 
But cannot set a new temperature yet.

**Reply by lbuijk** [reply]
> I have already managed to read them temp (both)! But cannot set a new temperature yet.

The SET temp is done with the first shell command. You can first try it from the prompt with a bash command.

**Reply by Josh004** [reply]
Everything works, it’s only the “parameters” of the set command:

shell_command:
  zet_temp_thermostaat: /usr/bin/curl -s -X POST -H "Session-token:$(/usr/bin/curl -s --request POST 'https://portal.icy.nl/login' --data 'username=USERNAME&password=PASSWORD' | awk -F":" '{print $13}' | awk -F'"' '{print $2}')" -d "temperature1=$(/usr/bin/curl -s -X GET -H "Authorization:Bearer INLOG TOKEN HOME ASSISTANT" -H "Content-Type:application/json" 'https://ip/ HOME ASSISTANT/api/states/input_number.zet_thermostaat_op' -k | awk -F":" '{print $3}' | awk -F'"' '{print $2}')" -d "uid=UID" --data 'username=USERNAME&password=PASSWORD' 'https://portal.icy.nl/data'

USERNAME = my username for ICY
PASSWORD = my password for ICY
INLOG TOKEN HOME ASSENT = ?
IP HOME ASSISTANT = 192.168.1.56  or external IP-Adres?

Thanks for all the hard work !!!

Kind greetings, Jos

**Reply by lbuijk** [reply]
> Everything works, it’s only the “parameters” of the set command:
> 
> shell_command: zet_temp_thermostaat: /usr/bin/curl -s -X POST -H "Session-token:$(/usr/bin/curl -s --request POST 'https://portal.icy.nl/login' --data 'username=USERNAME&password=PASSWORD' | awk -F":" '{print $13}' | awk -F'"' '{print Extra close brace or missing open brace $2}')" -d "temperature1=$(/usr/bin/curl -s -X GET -H "Authorization:Bearer INLOG TOKEN HOME ASSISTANT" -H "Content-Type:application/json" 'https://ip/ HOME ASSISTANT/api/states/input_number.zet_thermostaat_op' -k | awk -F":" '{print $3}' | awk -F'"' '{print $2}')" -d "uid=UID" --data 'username=USERNAME&password=PASSWORD' 'https://portal.icy.nl/data'
> 
> USERNAME = my username for ICY PASSWORD = my password for ICY INLOG TOKEN HOME ASSENT = ? IP HOME ASSISTANT = 192.168.1.56 or external IP-Adres?
> 
> Thanks for all the hard work !!!
> 
> Kind greetings, Jos

Inlog token home assistant = "Toegangstokens met lange levensduur" (when u use MFA).
IP Home assistant = LAN ip address

**Comment by gerard33** [comment]
Merged the PR anyway is it maybe solves some of the issues.

**Comment by Josh004** [comment]
Hi, must be little mistake, but still not right. What do I miss?shell_command:  zet_thermostaat_cv_thuis: /usr/bin/curl -s -X POST -H "Session-token:$(/usr/bin/curl -s --request POST 'https://portal.icy.nl/login' --data 'username=XXXXX&password=XXXXX' | awk -F":" '{print $13}' | awk -F'"' '{print $2}')" -d "temperature1=$(/usr/bin/curl -s -X GET -H "Authorization:Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6I.eyJpc3MiOiIzOThhMWQzMzE1ZjQ0Mzg2YTc0OWExYjExOGM5ZDRhMiIsImlhdCI6MTczOTUzMzUzOCwiZXhwIjoyMDU0ODkzNTM4fQ.BtzKHxHAwgZR1Du2BMsYzIwl5LcLBlRu8tzi6n6ArSU" -H "Content-Type:application/json" 'https://192.168.1.56/api/states/input_number.zet_thermostaat_cv_thuis_op' -k | awk -F":" '{print $3}' | awk -F'"' '{print $2}')" -d "uid=UID" --data 'username=XXXXX&password=XXXXX' 'https://portal.icy.nl/data'Met vriendelijke groet,JosOp 14 feb 2025 om 12:35 heeft lbuijk ***@***.***> het volgende geschreven:﻿

Everything works, it’s only the “parameters” of the set command:
shell_command: zet_temp_thermostaat: /usr/bin/curl -s -X POST -H "Session-token:$(/usr/bin/curl -s --request POST 'https://portal.icy.nl/login' --data 'username=USERNAME&password=PASSWORD' | awk -F":" '{print $13}' | awk -F'"' '{print Extra close brace or missing open brace $2}')" -d "temperature1=$(/usr/bin/curl -s -X GET -H "Authorization:Bearer INLOG TOKEN HOME ASSISTANT" -H "Content-Type:application/json" 'https://ip/ HOME ASSISTANT/api/states/input_number.zet_thermostaat_op' -k | awk -F":" '{print $3}' | awk -F'"' '{print $2}')" -d "uid=UID" --data 'username=USERNAME&password=PASSWORD' 'https://portal.icy.nl/data'
USERNAME = my username for ICY PASSWORD = my password for ICY INLOG TOKEN HOME ASSENT = ? IP HOME ASSISTANT = 192.168.1.56 or external IP-Adres?
Thanks for all the hard work !!!
Kind greetings, Jos

Inlog token home assistant = "Toegangstokens met lange levensduur" (when u use MFA).
IP Home assistant = LAN ip address

**Reply by lbuijk** [reply]
You must use your UID (replace "UID" with your number)

**Comment by Josh004** [comment]
Nope, still not working. I will try something else. Thanks for the effort. Met vriendelijke groet,JosOp 15 feb 2025 om 12:56 heeft lbuijk ***@***.***> het volgende geschreven:﻿
You must use your UID (replace "UID" with your number)
